### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Roland-Ntwali**
## In this PR, I did the following:

✔️ Create a class Classroom
✔️ Create a class Rental
✔️ Create the has-many/belongs-to relationship between Classroom and Student
✔️ Create a class Book
✔️ Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :person, :book
```
Comment (muneebulrehman) [inline rental.rb]: Both `person` and `book` instance variable should have a `getter` and a `setter`. However, in your case, you have defined `attr_reader` for them which only sets a `getter`. Kindly use `attr_accessor` to set both `getter` and `setter`.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @Roland-Ntwali , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Clean code.

## Required Changes ♻️
_Check the comments under the review._

- [ ] You are missing one of the requirements of the project where you need to set the relationship between `Person` and `Book` using the intermediate class `Rental`.  ⬇️ Kindly make the changes accordingly and then submit for a re-review.
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :person, :book, :date
```
Comment (1GORDON) [inline rental.rb]: Great job @Roland-Ntwali 🤝

- [ ] As the previous reviewer requested, kindly use `attr_accessor` to set both the getter and setter.

File: person.rb [hunk 3]
```diff
@@ -12,6 +12,7 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (1GORDON) [inline person.rb:17]: Great job @Roland-Ntwali 🤝

- [ ] [Optional] Kindly ensure the method for adding rentals should be present in the Person class.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Roland-Ntwali ✋

Good job so far!

## Highlights!
- Great use of Gitflow ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @Roland-Ntwali, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code based on previous reviewers' request 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**